### DISCUSSION TITLE
How do you use this plugin?

### DISCUSSION THREAD

**Discussion by arjunmahishi** [arjunmahishi/run-code.nvim#12] [General]
I want to understand how this plugin is being used. This will help me come up with a good roadmap for it. 

I mostly use it to automate commands that I run frequently. I have custom commands set up (in the telescope extensions) for running tests, building and copying binaries to the `PATH` running the current file directly, etc. This is what my custom command launcher looks like: 
![image](https://user-images.githubusercontent.com/11977524/192147920-4cd0ef0f-71a2-49a8-9c0b-a612898c92ce.png)

Tell me how you use it (or would like to use it) :)

**Comment by windowsrefund** [comment]
Hi

I really appreciate the flexible nature of your design and would like to use it to compliment a standard Terraform workflow. I went ahead and created a custom command of `terraform plan` in order to start getting a feel for things but wasn't expecting what I saw:

![flow-tfplan](https://github.com/user-attachments/assets/e2462482-4f82-4a74-acb4-6fcb34de490c)

Curious to hear thoughts on this. Thanks in advance.

**Reply by windowsrefund** [reply]
Oh wow. Looks like we spoke about this already in #9 . Actually, that was related to the input side of things where this is about the output of a custom command.

I just edited the custom command to `terraform plan -no-color` which resulted in something quite usable. That said, I'm certainly curious to know why the output appears to have "scrolled up" in such a way as to leave roughly 50% of the vertical space empty?

**Reply by arjunmahishi** [reply]
Hi @windowsrefund, Thank you for the kind words!

`terraform plan -no-color` seems like a viable fix for the unescaped characters in the output. Because we are displaying the out as plain text in a buffer. We could potentially explore streaming the output to a terminal buffer instead of a regular text based buffer. That might solve most of the formatting, scrolling problems.

> output appears to have "scrolled up" 

This should normally not  happen. Because the plugin just appends lines written to STDOUT/STDERR to the buffer. It could be that the terraform command is dynamically updating lines in STDOUT (to update progress inline). This messes up the output buffer. Like I mentioned above, streaming the output to a terminal buffer would fix this.

I will look into this. Thanks!

**Reply by windowsrefund** [reply]
Well, well.... I found the culprit. I had an autocmd in place which centers my cursor by running `normal! zz` on every `CursorMoved` event. The net effect of the autocmd is that my cursor is kept in the middle of the screen at all times.

Removing it results in things working as intended.

**Reply by arjunmahishi** [reply]
ohh wow, great find!!
